packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -99,10 +106,54 @@ const FIRST_DEVICE_ID = 7;
 const FIRST_OBJECT_ID = 20;
 const FIRST_DEVNODE_ID = 30;
 const FIRST_PROMISE_ID = 40;
+const FIRST_CRANK_NUMBER = 0;
+
+const makeCounter = (storage, key, initial = fromStr(storage.get(key))) => {
+  initial = natNum(initial);
+  const setStorage = id => storage.set(key, toStr(id));
+  const getStorage = () => {
+    if (!storage.has(key)) {
+      throw new Error(`storage lacks required key ${key}`);
+    }
+    return fromStr(storage.get(key));
+  };
+  /**
+   * @typedef {Object} Counter
+   * @property {() => void} createStartingState
+   * @property {() => bigint} getNextID
+   * @property {() => Array<bigint>} getAllIDs
+   * @property {() => bigint} allocateNextID
+   */
+  const counter = harden({
+    createStartingState: () => setStorage(initial),
```

Comment:
Perhaps a matter of taste, but I really don't care for separating `createStartingState` from `makeCounter`, especially since the initialization value is a parameter to the latter.

Reply:
Ah good point.